astropy/convolution/convolve.py

```diff
@@ -364,16 +370,17 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
         new_result._separable = passed_array._separable
         if isinstance(passed_kernel, Kernel):
             new_result._separable = new_result._separable and passed_kernel._separable
-        return new_result
+        return new_result * passed_quantity
     elif array_dtype.kind == 'f':
         # Try to preserve the input type if it's a floating point type
         # Avoid making another copy if possible
+        result *= passed_quantity
```

Comment:
Does this make a copy, and thus contradict the comment above it?

Reply:
Thanks for the review! I agree the multiplication by 1 looks redundant - the only alternative I can see is adding a conditional to each return statement, unless there's another way? As for the *=, I wasn't aware that created another copy?

Reply:
I am not sure either. I think that depends on the implementation details in `__rmul__`.

Reply:
`*=` should be inplace, so I'm not worried about this, but line 373 above might create a copy.

Returning a ternary (`new_result * passed_quantity if <passed_quantity is a unit> else new_result`) might be better than this unneeded multiplication operation